web/packages/teleterm/src/ui/components/OfflineGateway.tsx

```diff
@@ -21,71 +21,77 @@ import { ButtonPrimary, Flex, Text } from 'design';
 
 import * as Alerts from 'design/Alert';
 import Validation from 'shared/components/Validation';
-
-import LinearProgress from 'teleterm/ui/components/LinearProgress';
-
-import { DocumentGatewayProps } from '../DocumentGateway/DocumentGateway';
+import { Attempt } from 'shared/hooks/useAsync';
 
 import { PortFieldInput } from './FieldInputs';
 
-type OfflineDocumentGatewayProps = Pick<
-  DocumentGatewayProps,
-  'connectAttempt' | 'defaultPort' | 'reconnect'
->;
+export function OfflineGateway(props: {
+  connectAttempt: Attempt<void>;
+  /** Setting `isSupported` to false hides the port input. */
+  gatewayPort:
+    | { isSupported: true; defaultPort: string }
+    | { isSupported: false };
+  reconnect(port?: string): void;
+  /** Gateway target displayed in the UI, for example, 'cockroachdb'. */
+  targetName: string;
+  /** Gateway kind displayed in the UI, for example, 'database'. */
+  gatewayKind: string;
+}) {
+  const defaultPort = props.gatewayPort.isSupported
+    ? props.gatewayPort.defaultPort
+    : undefined;
+
+  const [port, setPort] = useState(defaultPort);
+  const [reconnectRequested, setReconnectRequested] = useState(false);
 
-export function OfflineGateway(props: OfflineDocumentGatewayProps) {
-  const [port, setPort] = useState(props.defaultPort);
-  const statusDescription =
-    props.connectAttempt.status === 'processing' ? 'being set up' : 'offline';
   const isProcessing = props.connectAttempt.status === 'processing';
-  const shouldShowPortInput =
-    props.connectAttempt.status === 'error' || port !== props.defaultPort;
+  const statusDescription = isProcessing ? 'being set up…' : 'offline.';
+  const shouldShowReconnectControls =
+    props.connectAttempt.status === 'error' || reconnectRequested;
 
   return (
-    <Flex
-      maxWidth="680px"
-      width="100%"
-      flexDirection="column"
-      mx="auto"
-      alignItems="center"
-      mt={11}
-    >
-      <Text
-        typography="h5"
-        color="text.main"
-        mb={2}
-        style={{ position: 'relative' }}
-      >
-        The database connection is {statusDescription}
-        {/* TODO(ravicious): Use doc.status instead of LinearProgress. */}
-        {props.connectAttempt.status === 'processing' && <LinearProgress />}
+    <Flex flexDirection="column" m="auto" alignItems="center" maxWidth="500px">
```

Comment:
I'm not sure if we should center it. We don't really have any other documents that do this, do we? When the app is almost fullscreen, especially on bigger screens, the transition from a centered loading state to a success state that's aligned to the top is going to look a bit weird.

Reply:
We have a very similar screen in `DocumentCluster` where we center the content. 
But I think you are right, we should rather align it to the top. I changed it in both places.

Now it looks like below:
<img width="1419" alt="image" src="https://github.com/gravitational/teleport/assets/20583051/8de3b4a9-0cb1-4fb0-a330-94c2f379b8cf">